### ISSUE TITLE
Cannot reproduce Table 7 X5 Qini scores: S Learner and T Learner significantly off

### ISSUE THREAD

**Issue by onder-prosus** [vdblm/CausalPFN#8]
## Summary

Running the paper's own code (`notebooks/qini.ipynb` pipeline, `benchmarks/baselines/econml.py`, `benchmarks/retail_hero.py`) on the full X5 dataset with `seed=82718` and `hpo=False` does **not** reproduce the normalized Qini scores reported in Table 7.

S Learner and T Learner are significantly off, while DA Learner and X Learner match.

## Environment

All packages installed via `pip install -e '.[dev]'` from the repo, matching the paper's pinned versions:

- Python 3.10.13
- econml 0.15.1
- scikit-learn 1.5.2
- scikit-uplift 0.5.1
- pandas 2.3.0
- numpy 1.26.4
- Platform: Linux (Ubuntu), CPU-only

## Reproduction Script

```python
import sys, time, numpy as np
sys.path.insert(0, "/path/to/CausalPFN")

from benchmarks.retail_hero import X5Dataset
from benchmarks.baselines.econml import (
    TLearnerBaseline, SLearnerBaseline, XLearnerBaseline,
    DALearnerBaseline, ForestDRLearnerBaseline,
)
from src.causalpfn.evaluation import get_qini_curve

SEED = 82718
dataset = X5Dataset(seed=SEED, n_folds=5)

baselines = {
    "T Learner":         TLearnerBaseline(hpo=False),
    "S Learner":         SLearnerBaseline(hpo=False),
    "X Learner":         XLearnerBaseline(hpo=False),
    "DA Learner":        DALearnerBaseline(hpo=False),
    "Forest DR Learner": ForestDRLearnerBaseline(hpo=False),
}

fold_preds = {name: [] for name in baselines}
for fold_idx in range(5):
    fold = dataset[fold_idx]
    for name, baseline in baselines.items():
        cate = baseline.estimate_cate(
            X_train=fold.X_train, t_train=fold.t_train,
            y_train=fold.y_train, X_test=fold.X_test,
        )
        fold_preds[name].append((fold.t_test, fold.y_test, cate))

raw_scores = {}
for name, preds in fold_preds.items():
    all_t = np.concatenate([p[0] for p in preds])
    all_y = np.concatenate([p[1] for p in preds])
    all_cate = np.concatenate([p[2] for p in preds])
    _, q = get_qini_curve(all_t, all_y, all_cate, normalize=True)
    raw_scores[name] = q

best = max(raw_scores.values())
for name, q in sorted(raw_scores.items(), key=lambda x: -x[1]):
    print(f"  {name:<22} {q/best:.3f}  (raw={q:.6f})")
```

## Results

**Our output (normalized, top=1.000):**

| Method | Ours | Paper Table 7 X5 | Match? |
|---|---|---|---|
| DA Learner | **1.000** (raw=0.8209) | 0.969 | ✅ OK |
| X Learner | 0.976 (raw=0.8012) | 0.965 | ✅ OK |
| Forest DR Learner | 0.925 (raw=0.7591) | (not in Table 7) | — |
| T Learner | 0.782 (raw=0.6423) | 0.953 | ❌ MISMATCH |
| S Learner | 0.701 (raw=0.5754) | **1.000** | ❌ MISMATCH |

**Per-fold breakdown (not normalized):**

| Method | Fold 1 | Fold 2 | Fold 3 | Fold 4 | Fold 5 |
|---|---|---|---|---|---|
| T Learner | 0.548 | 0.680 | 0.674 | 0.716 | 0.613 |
| S Learner | 0.567 | 0.541 | 0.628 | 0.613 | 0.592 |
| X Learner | 0.788 | 0.733 | 0.895 | 0.831 | 0.774 |
| DA Learner | 0.780 | 0.726 | 0.886 | 0.870 | 0.861 |
| Forest DR | 0.711 | 0.791 | 0.746 | 0.778 | 0.770 |

S Learner is **last in every single fold** with scores barely above random (0.50). The paper reports it as the **best** method.

## Key Observations

1. **DA Learner and X Learner match the paper well** — their implementation is likely correct.
2. **S Learner is consistently the worst** (avg ~0.585) despite being reported as best (1.000) in Table 7. The `hpo=False` S Learner uses `RandomForestRegressor(n_estimators=100, max_depth=6, min_samples_leaf=~1600)`. With such shallow trees and no treatment-specific branching, the forest likely ignores the treatment indicator appended to X, yielding near-random CATE estimates.
3. **T Learner is also significantly lower** (0.782 normalized vs 0.953 paper).

## Question

Was Table 7 perhaps generated with `hpo=True` for S Learner and T Learner specifically, or with different hyperparameters than what `hpo=False` currently uses? The `qini.ipynb` notebook hardcodes `hpo=False` for all baselines, but perhaps the actual Table 7 run used a different configuration.

Any guidance on reproducing the exact Table 7 numbers would be appreciated.
